### PR TITLE
add mod push down and fix copr-test

### PR DESCRIPTION
**Pull request by Yisaer**
Signed-off-by: yisaer <disxiaofei@163.com>

add push down functions according to https://github.com/pingcap/tidb/pull/31983

### CODE REVIEW

**Review by hawkingrei — APPROVED** [review]

File: push-down-test/sql/randgen-topn/1_arith_1.sql [hunk 1]
```diff
@@ -176,7 +176,7 @@
...
- SELECT ( `col_varchar_64_key` * `col_enum_key` ) % `col_tinyint_key` AS field1 FROM `table10_int_autoinc` WHERE '2028-09-11' + `col_decimal_unsigned` ORDER BY field1 LIMIT 9 /* QNO 180 CON_ID 92 */ ;
+--  SELECT ( `col_varchar_64_key` * `col_enum_key` ) % `col_tinyint_key` AS field1 FROM `table10_int_autoinc` WHERE '2028-09-11' + `col_decimal_unsigned` ORDER BY field1 LIMIT 9 /* QNO 180 CON_ID 92 */ ;
```
Comment (Yisaer) [inline push-down-test/sql/randgen-topn/1_arith_1.sql:179]: I removed some sql in tests due to the generated results are out of range.

File: push-down-test/sql/randgen-topn/1_arith_2.sql [hunk 2]
```diff
@@ -294,7 +294,7 @@
...
  SELECT ( 'i' / `col_tinyint_unsigned` ) * ( ( 'p' DIV 4140778382390460416 ) * `col_smallint_key` ) AS field1 FROM `table0_int_autoinc` WHERE 'hjjironhbfuqudkhutkeretgvzmpezczyydjzebmimcmyvmaoieejlggbvspoypxbwbglltfzkhuzaedwgbertojtjwfevviclcuvvtesjdkxfqwmcqejezuoiggvnpmdtggrahqlblziabpagsdndzuctfxwqpybuyivyxyhiphxzweljystxplssxhquofmdalqzzksonifzjynaizuqdo' MOD ( '1989-04-10' % 4500 ) ORDER BY field1 LIMIT 1 /* QNO 296 CON_ID 188 */ ;
  SELECT '1984-04-02 15:48:12.006100' MOD 'c' AS field1 FROM `table0_int_autoinc`  ORDER BY field1 LIMIT 7 /* QNO 297 CON_ID 188 */ ;
- SELECT ( 'u' * ( `col_char_255_key` - `col_varchar_1` ) ) * '1995-09-22' AS field1 FROM `table20_int_autoinc` WHERE 5436689175166320640 + `col_bit` ORDER BY field1 LIMIT 9 /* QNO 298 CON_ID 188 */ ;
+ SELECT ABS( ( 'u' * ( `col_char_255_key` - `col_varchar_1` ) ) ) * '1995-09-22' AS field1 FROM `table20_int_autoinc` WHERE 5436689175166320640 + `col_bit` ORDER BY field1 LIMIT 9 /* QNO 298 CON_ID 188 */ ;
```
Comment (Yisaer) [inline push-down-test/sql/randgen-topn/1_arith_2.sql:297]: Adding `ABS` due to it will case failure by comparing `0` and `-0`

**Review by disksing — APPROVED** [review]

**Pull request merged**